Workaround race condition on default sa creation

jorhett opened the pull request:
This is a workaround for a known race condition where you can try to add a Pod before the cluster has created the default serviceaccount https://github.com/kubernetes/kubernetes/issues/66689
```
Notice: /Stage[main]/Kubernetes::Cluster_roles/Kubernetes::Kubeadm_init[kube-master]/Exec[kubeadm init]/returns: executed successfully
Notice: /Stage[main]/Kubernetes::Kube_addons/Exec[Install cni network provider]/returns: serviceaccount/weave-net created
Notice: /Stage[main]/Kubernetes::Kube_addons/Exec[Install cni network provider]/returns: clusterrole.rbac.authorization.k8s.io/weave-net created
Notice: /Stage[main]/Kubernetes::Kube_addons/Exec[Install cni network provider]/returns: clusterrolebinding.rbac.authorization.k8s.io/weave-net created
Notice: /Stage[main]/Kubernetes::Kube_addons/Exec[Install cni network provider]/returns: role.rbac.authorization.k8s.io/weave-net created
Notice: /Stage[main]/Kubernetes::Kube_addons/Exec[Install cni network provider]/returns: rolebinding.rbac.authorization.k8s.io/weave-net created
Notice: /Stage[main]/Kubernetes::Kube_addons/Exec[Install cni network provider]/returns: daemonset.extensions/weave-net created
Notice: /Stage[main]/Kubernetes::Kube_addons/Exec[Install cni network provider]/returns: executed successfully
Notice: /Stage[main]/Profile::Kubectl::Templates/Exec[kubectl_template_busybox]/returns: Error from server (Forbidden): error when creating "/k8s/templates/busybox.yaml": pods "busybox" is forbidden: error looking up service account default/default: serviceaccount "default" not found
Error: 'kubectl apply -f /k8s/templates/busybox.yaml' returned 1 instead of one of [0]
Error: /Stage[main]/Profile::Kubectl::Templates/Exec[kubectl_template_busybox]/returns: change from 'notrun' to ['0'] failed: 'kubectl apply -f /k8s/templates/busybox.yaml' returned 1 instead of one of [0]
```

This adds a test for the SA, which will trigger the accounting query to speed up its creation.

davejrt commented on manifests/kubeadm_init.pp:
```diff
@@ -25,4 +25,12 @@
 
+  # This prevents a known race condition https://github.com/kubernetes/kubernetes/issues/66689
+  exec { 'wait for default serviceaccount creation':
+    command     => 'kubectl -n default get serviceaccount default -o name'
```
might need to put a comma on the end here

jorhett replied:
Fixed.

jorhett commented:
I've restructured so it as a new defined type so it can be used with other namespaces.

In doing so I realized that the exec path was (A) marked optional everywhere and it's not, and (B) hardcoded in the cluster file. I've moved this out to a module parameter and adjusted the things which use it.

davejrt commented:
LGTM, thanks for the PR.
